### PR TITLE
cilium: Change Consumable LabelList to LabelArray - fix #838

### LINKED ISSUE #838: Change Consumable's LabelList to a LabelArray
Per [this comment](https://github.com/cilium/cilium/pull/815#discussion_r118845655) we should update Consumable's LabelList to be a LabelArray.

### PR DESCRIPTION
**Pull request by stevenceuppens**
Signed-off-by: Steven Ceuppens <steven.ceuppens@icloud.com>

### CODE REVIEW

File: pkg/policy/consumer.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ type Consumable struct {
 	Mutex sync.RWMutex
 	// Labels are the Identity of this consumable
 	Labels *Identity `json:"labels"`
-	// LabelList contains the same labels from identity in a form of a list, used for faster lookup
-	LabelList []*labels.Label `json:"-"`
+	// LabelArray contains the same labels from identity in a form of a list, used for faster lookup
+	LabelArray []*labels.Label `json:"-"`
```
Comment (aanm) [inline pkg/policy/consumer.go]: I think what @ianvernon meant in #838 was that `LabelList []*labels.Label` should be replaced with `LabelArray labels.LabelArray`

Comment (ianvernon) [reply pkg/policy/consumer.go]: Yes, @aanm is correct.

**Comment by stevenceuppens** [comment]
@aanm I updated according your request;

I also found other occurrences of []*labels.Label that could be converted to labels.LabelArray

Do i revert the LabelList to LabelArray change i made first? or do we keep this for consistency?
other parts of the source use labels as symbol name; but this is conflicting in Comsumable struct.

**Comment by tgraf** [comment]
> Do i revert the LabelList to LabelArray change i made first? or do we keep this for consistency?
> other parts of the source use labels as symbol name; but this is conflicting in Comsumable struct.

Looks good, can you squash the two commits together wit git rebase?

**Comment by tgraf** [comment]
> Do i revert the LabelList to LabelArray change i made first

I think we can keep the variable name as it was and only change type. You can make the change on top as a separate commit and then just squash all three together to a single change.

**Comment by tgraf** [comment]
@stevenceuppens Actually, this is fine as well as-is. I like it better than `LabelList`

**Review by tgraf — APPROVED** [review]

**Comment by stevenceuppens** [comment]
Ok, commits are squashed!

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
Merged, thanks @stevenceuppens!

**Pull request merged**